gateway-ha/src/test/java/io/trino/gateway/ha/TestPutRequestHandler.java

```diff
@@ -0,0 +1,120 @@
+import static org.assertj.core.api.Assertions.assertThat;
+
+@TestInstance(Lifecycle.PER_CLASS)
+public class TestPutRequestHandler
+{
+    private final OkHttpClient httpClient = new OkHttpClient();
+    private MockWebServer mockTrinoServer = new MockWebServer();;
+
+    private final int routerPort = 21001 + (int) (Math.random() * 1000);
+    private final int customBackendPort = 21000 + (int) (Math.random() * 1000);
+
+    private final static String CUSTOM_PUT_ENDPOINT = "/v1/custom"; // this is enabled in test-config-template.yml
+    private final static String HEALTH_CHECK_ENDPOINT = "/v1/info";
+    private final static String PUT = "PUT";
+
+    @BeforeAll
+    public void setup()
+            throws Exception
+    {
+        HaGatewayTestUtils.prepareMockBackend(mockTrinoServer, customBackendPort, "default custom response");
+        mockTrinoServer.setDispatcher(new Dispatcher() {
+            @Override
+            public MockResponse dispatch(RecordedRequest request)
+            {
+                if (request.getPath().equals(HEALTH_CHECK_ENDPOINT)) {
+                    return new MockResponse().setResponseCode(200)
+                            .setHeader(CONTENT_TYPE, JSON_UTF_8)
+                            .setBody("{\"starting\": false}");
+                }
+
+                if (request.getMethod().equals(PUT) && request.getPath().equals(CUSTOM_PUT_ENDPOINT)) {
+                    return new MockResponse().setResponseCode(200)
+                            .setHeader(CONTENT_TYPE, JSON_UTF_8)
+                            .setBody("ok");
+                }
+
+                return new MockResponse().setResponseCode(404);
+            }
+        });
+
+        // seed database
+        HaGatewayTestUtils.TestConfig testConfig =
+                HaGatewayTestUtils.buildGatewayConfigAndSeedDb(routerPort, "test-config-template.yml");
+
+        // Start Gateway
+        String[] args = {testConfig.configFilePath()};
+        HaGatewayLauncher.main(args);
+
+        // Set up backend
+        HaGatewayTestUtils.setUpBackend(
+                "custom", "http://localhost:" + customBackendPort, "externalUrl", true, "adhoc", routerPort);
+    }
+
+    @AfterAll
+    public void cleanup()
+            throws Exception
+    {
+        if (mockTrinoServer != null) {
+            mockTrinoServer.shutdown();
+        }
+    }
+
+    @Test
+    public void testPutRequestHandler()
+            throws Exception
+    {
+        RequestBody requestBody =
+                RequestBody.create(MediaType.parse("application/json; charset=utf-8"), "SELECT 1");
+        Request putRequest =
+                new Request.Builder()
+                        .url("http://localhost:" + routerPort + CUSTOM_PUT_ENDPOINT)
+                        .put(requestBody)
+                        .build();
+
+        try (Response response = httpClient.newCall(putRequest).execute()) {
+            assertThat(response.isSuccessful()).isTrue();
```

Comment:
check that the response body is equal to the body in your mock (`ok`). You can extract this to a constant field

Reply:
addressed in [c58ac48](https://github.com/trinodb/trino-gateway/pull/543/commits/c58ac4863a8d30c5754a3127fb98de4fc88b47d3)